python_bindings/test/correctness/memoize.py

```diff
@@ -0,0 +1,24 @@
+from halide import Func, Var
+
+
+def test_memoize():
+    x = Var("x")
+
+    f = Func("f")
+    f[x] = 0.0
+    f[x] = f[x] + 1
+    f.memoize()
+
+    output = Func("output")
```

Comment:
Did you want an f.compute_root() here? I don't think much caching is taking place otherwise

Reply:
An inlined, memoized function should throw a warning...